templates/zerver/register.html

```diff
@@ -118,8 +118,26 @@ <h1>{{ _('Create your account') }}</h1>
                 <div class="input-box no-validation">
                     <input type='hidden' name='key' value='{{ key }}' />
                     <input type='hidden' name='timezone' id='timezone'/>
+                    <input type="hidden" name="email_address_visibility" id="email_address_visibility"/>
                     <label for="id_email" class="inline-block label-title">{{ _('Email') }}</label>
                     <div id="id_email">{{ email }}</div>
+                    {% if not creating_new_team %}
+                    <p id="new-user-email-address-visibility">
+                        <span class="current-selected-option">
+                            {% if default_email_address_visibility == email_address_visibility_admins_only %}
+                                {% trans %}Administrators of this Zulip organization will be able to see this email address.
+                                {% endtrans %}
+                            {% elif default_email_address_visibility == email_address_visibility_nobody %}
+                                {% trans %}Nobody in this Zulip organization will be able to see this email address.
+                                {% endtrans %}
+                            {% else %}
+                                {% trans %}Other users in this Zulip organization will be able to see this email address.
+                                {% endtrans %}
+                            {% endif %}
```

Comment:
Reading these rules, maybe it's worth having an extra case for "Moderators of this ..." -- I can see that being meaningfully different for some users than "Other users" from a privacy standpoint.

(I don't think the guest/members/full members distinctions for this sort of setting are likely to be meaningful for a new user).